plugin/trino-phoenix-5/src/main/java/io/trino/plugin/phoenix/PhoenixClientModule.java

```diff
@@ -0,0 +1,179 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.phoenix;
+
+import com.google.inject.Binder;
+import com.google.inject.Key;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.airlift.log.Logger;
+import io.trino.plugin.base.classloader.ClassLoaderSafeConnectorMetadata;
+import io.trino.plugin.base.classloader.ClassLoaderSafeConnectorPageSinkProvider;
+import io.trino.plugin.base.classloader.ClassLoaderSafeConnectorSplitManager;
+import io.trino.plugin.base.classloader.ForClassLoaderSafe;
+import io.trino.plugin.base.util.LoggingInvocationHandler;
+import io.trino.plugin.jdbc.ConnectionFactory;
+import io.trino.plugin.jdbc.DriverConnectionFactory;
+import io.trino.plugin.jdbc.ForwardingJdbcClient;
+import io.trino.plugin.jdbc.JdbcClient;
+import io.trino.plugin.jdbc.JdbcMetadataConfig;
+import io.trino.plugin.jdbc.JdbcMetadataSessionProperties;
+import io.trino.plugin.jdbc.JdbcPageSinkProvider;
+import io.trino.plugin.jdbc.JdbcRecordSetProvider;
+import io.trino.plugin.jdbc.MaxDomainCompactionThreshold;
+import io.trino.plugin.jdbc.TypeHandlingJdbcConfig;
+import io.trino.plugin.jdbc.TypeHandlingJdbcSessionProperties;
+import io.trino.plugin.jdbc.credential.EmptyCredentialProvider;
+import io.trino.plugin.jdbc.jmx.StatisticsAwareConnectionFactory;
+import io.trino.plugin.jdbc.jmx.StatisticsAwareJdbcClient;
+import io.trino.spi.TrinoException;
+import io.trino.spi.connector.ConnectorMetadata;
+import io.trino.spi.connector.ConnectorPageSinkProvider;
+import io.trino.spi.connector.ConnectorRecordSetProvider;
+import io.trino.spi.connector.ConnectorSplitManager;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.phoenix.jdbc.PhoenixDriver;
+import org.apache.phoenix.jdbc.PhoenixEmbeddedDriver;
+
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.reflect.Reflection.newProxy;
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.trino.plugin.jdbc.JdbcModule.bindSessionPropertiesProvider;
+import static io.trino.plugin.jdbc.JdbcModule.bindTablePropertiesProvider;
+import static io.trino.plugin.phoenix.PhoenixErrorCode.PHOENIX_CONFIG_ERROR;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.weakref.jmx.guice.ExportBinder.newExporter;
+
+public class PhoenixClientModule
```

Comment:
Can I suggest to split the commit into two commits?

1. sth like "Make a verbatim copy of trino-phoenix module as trino-phoenix-5". This commit message should also include the commit hash from which you're copying from
2. all the other changes -- pom changes, minor code changes, docs changes and packaging

My intention is 

- to make it clear in `git blame` where the code comes from, just in case someone needs to track something
  - today it's kind of easy, but one day the Phoenix 4 module will have been removed and this commit will appear to be _the_ addition of the Phoenix module
- make it easier to track changes in individual modules and perhaps cross-apply them if needed (eg #6807)

This can be done quite safely in git with
```
git checkout master
cp trino-phoenix ..
git add trino-phoenix-5
git commit ...

git checkout your-branch
git reset master # no --hard here
git add . 
git status # make sure to add any new files (but i don't think there are any)
git commit

# sanity check that the end result is the same as the last pushed out version
git diff @{u}
```

Reply:
Hmm... Good call. Just so I understand... Why would the first part need to be done on the master branch? (and I would not have the right to do that).

Would you do the first part? (The copy and git add ...)
When that's done I'd make the changes for Phoenix 5.

Right?

Reply:
Or did you mean to say that this is how I can fix my branch?
If that, it's really not a problem to start with a new PR.

Reply:
I meant this within your PR. And you don't need to create a new PR for that. I hope this isn't a big trouble for you, but let me know if i can help.

Reply:
oh, i see now you did this already. thanks!